### PR TITLE
SAK-42474 Sakai grader

### PR DESCRIPTION
**Pull request by adrianfish**
https://jira.sakaiproject.org/browse/SAK-42474

Adds an alternative way of grading assignment submissions, hopefully
making the process more fluid.

### CODE REVIEW

**Comment by adrianfish** [comment]
@mpellicer  Fixed, no biggy.

File: webcomponents/tool/src/main/frontend/sakai-element.js [hunk 1]
```diff
@@ -22,6 +22,36 @@ class SakaiElement extends LitElement {
     // Pass the call on to the imported function
     return loadProperties(options);
   }
+
+  messageReceived(message) {
```
Comment (ern) [inline webcomponents/tool/src/main/frontend/sakai-element.js]: Not used? might be able to remove

File: webcomponents/tool/src/main/frontend/sakai-file-picker.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  reset() {
+
+    this.files = [];
+    this.requestUpdate();
+  }
+}
+
+customElements.define("sakai-file-picker", SakaiFilePicker);
```
Comment (ern) [inline webcomponents/tool/src/main/frontend/sakai-file-picker.js]: does the element sakai-grader-file-picker make sense?

File: assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentServiceConstants.java [hunk 3]
```diff
@@ -115,6 +115,30 @@
     // SAK-33971 Property for enable/disable the group filter
     public static final String PROP_ASSIGNMENT_GROUP_FILTER_ENABLED = "assignment.group.filter.enabled";
 
+
+    /**
+     * The configuration choice of with grading option or not
+     */
+    public static final String WITH_GRADES = "with_grades";
```
Comment (ern) [inline assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentServiceConstants.java]: Are these duplicates?

File: assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentServiceConstants.java [hunk 4]
```diff
@@ -115,6 +115,30 @@
...
+
+    public static final String GRADE_SUBMISSION_DONT_CLEAR_CURRENT_ATTACHMENTS = "grade_submission_dont_clear_current_attachments";
+
+    // submission level of resubmit due time
+    public static final String ALLOW_RESUBMIT_CLOSEYEAR = "allow_resubmit_closeYear";
+    public static final String ALLOW_RESUBMIT_CLOSEMONTH = "allow_resubmit_closeMonth";
+    public static final String ALLOW_RESUBMIT_CLOSEDAY = "allow_resubmit_closeDay";
+    public static final String ALLOW_RESUBMIT_CLOSEHOUR = "allow_resubmit_closeHour";
+    public static final String ALLOW_RESUBMIT_CLOSEMIN = "allow_resubmit_closeMin";
+
```
Comment (ern) [inline assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentServiceConstants.java]: Are these duplicates?

**Pull request merged**